samples/ANCMStressTestSample/Program.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Server.IISIntegration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace ANCMStressTestApp
+{
+    public class Program
+    {
+        public static IApplicationLifetime AppLifetime;
+        public static bool AppLifetimeStopping = false;
+
+        public static void Main(string[] args)
+        {
+            var host = new WebHostBuilder()
+                .ConfigureLogging((_, factory) =>
+                {
+                    factory.AddConsole();
+                })
+                .UseKestrel()
+                .UseStartup<Startup>()
+                .Build();
+
+            AppLifetime = (IApplicationLifetime)host.Services.GetService(typeof(IApplicationLifetime));
+            AppLifetime.ApplicationStopping.Register(
+                () => {
+                    AppLifetimeStopping = true;
```

Comment:
Having this been a bool seems racy to me. Provide a cancellation token that will be canceled instead? Also you can pass in that cancellation token to Send/ReceiveAsync

Reply:
This was copied from Jeong Hwan's original test app for functional test. Yes, for stress test purpose, we need to handle application shutdown more carefully. Made a change to how we handle close handshake.